src/main/java/appeng/block/grindstone/GrinderBlock.java

```diff
@@ -47,11 +46,11 @@ public ActionResultType onActivated(final World w, final BlockPos pos, final Pla
             final @Nullable ItemStack heldItem, final BlockRayTraceResult hit) {
         final GrinderTileEntity tg = this.getTileEntity(w, pos);
         if (tg != null && !InteractionUtil.isInAlternateUseMode(p)) {
-            if (p instanceof ServerPlayerEntity) {
```

Comment:
is p nullable and this elides a null-check? I don't remember

Reply:
`InteractionUtil.isInAlternateUseMode(p)` guards against it by throwing a NPE before it even reaches the `instanceof`.
But otherwise no idea. Looks like an outlier. There is also no special handler here for fakeplayers, that is just for the crank.
Otherwise anything else follows the pattern with `getTile && !isRemote`